### PR TITLE
Allow document walks in `customElements.define` to be delayed.

### PR DESCRIPTION
**Pull request by bicknellr**
.. Is this the expected API for this feature?

### CODE REVIEW

File: src/CustomElementRegistry.js [hunk 1]
```diff
@@ -99,14 +111,98 @@ export default class CustomElementRegistry {
...
+      if (!this._flushScheduled) {
+        this._flushScheduled = true;
+        this._flushCallback(() => this._doFlush());
+      }
+    }
+  }
+
+  _doFlush() {
+    // If no new definitions were defined, don't attempt to flush.
+    if (this._flushScheduled === false) return;
```
Comment (kevinpschaaf) [inline src/CustomElementRegistry.js]: This case should not be possible, yeah?

File: src/CustomElementRegistry.js [hunk 2]
```diff
@@ -99,14 +111,98 @@ export default class CustomElementRegistry {
...
+   *
+   * // Setting the flush callback to `undefined` walks the document for
+   * // upgrades and resolves any pending `whenDefined` promises, if any calls
+   * // to `define` were made since the last flush.
+   * customElements.polyfillSetFlushCallback(undefined);
+   * ```
+   *
+   * @param {Function} flushCallback
+   */
+  polyfillSetFlushCallback(flushCallback) {
```
Comment (kevinpschaaf) [inline src/CustomElementRegistry.js]: Let's make this a get/set `polyfillFlushCallback` accessor, and make it match the `priorCustomElements` interface.

File: src/CustomElementRegistry.js (hunk 1 above)
Comment (bicknellr) [inline src/CustomElementRegistry.js]: If you hold on to one of the functions and call it multiple times, possibly.

File: src/CustomElementRegistry.js (hunk 2 above)
Comment (bicknellr) [inline src/CustomElementRegistry.js]: Fixed.

**Pull request merged**